src/fadec/a320_fadec/src/EngineControl.h

```diff
@@ -806,48 +808,53 @@ class EngineControl {
-                                      &fuelCenter);
-        SimConnect_SetDataOnSimObject(hSimConnect, DataTypesID::FuelLeftMain, SIMCONNECT_OBJECT_ID_USER, 0, 0, sizeof(double), &fuelLeft);
-        SimConnect_SetDataOnSimObject(hSimConnect, DataTypesID::FuelRightMain, SIMCONNECT_OBJECT_ID_USER, 0, 0, sizeof(double), &fuelRight);
-        SimConnect_SetDataOnSimObject(hSimConnect, DataTypesID::FuelLeftAux, SIMCONNECT_OBJECT_ID_USER, 0, 0, sizeof(double), &fuelLeftAux);
-        SimConnect_SetDataOnSimObject(hSimConnect, DataTypesID::FuelRightAux, SIMCONNECT_OBJECT_ID_USER, 0, 0, sizeof(double),
-                                      &fuelRightAux);
-      }
+
+      fuelLeft = (fuelLeftPre / fuelWeightGallon);          // USG
+      fuelRight = (fuelRightPre / fuelWeightGallon);        // USG
+      fuelCenter = (fuelCenterPre / fuelWeightGallon);      // USG
+      fuelLeftAux = (fuelAuxLeftPre / fuelWeightGallon);    // USG
+      fuelRightAux = (fuelAuxRightPre / fuelWeightGallon);  // USG
+
+      SimConnect_SetDataOnSimObject(hSimConnect, DataTypesID::FuelCenterMain, SIMCONNECT_OBJECT_ID_USER, 0, 0, sizeof(double), &fuelCenter);
+      SimConnect_SetDataOnSimObject(hSimConnect, DataTypesID::FuelLeftMain, SIMCONNECT_OBJECT_ID_USER, 0, 0, sizeof(double), &fuelLeft);
+      SimConnect_SetDataOnSimObject(hSimConnect, DataTypesID::FuelRightMain, SIMCONNECT_OBJECT_ID_USER, 0, 0, sizeof(double), &fuelRight);
+      SimConnect_SetDataOnSimObject(hSimConnect, DataTypesID::FuelLeftAux, SIMCONNECT_OBJECT_ID_USER, 0, 0, sizeof(double), &fuelLeftAux);
+      SimConnect_SetDataOnSimObject(hSimConnect, DataTypesID::FuelRightAux, SIMCONNECT_OBJECT_ID_USER, 0, 0, sizeof(double), &fuelRightAux);
     } else if (!uiFuelTamper && refuelStartedByUser == 1) {  // Detects refueling from the EFB
-      simVars->setFuelLeftPre(leftQuantity);                 // in LBS
-      simVars->setFuelRightPre(rightQuantity);               // in LBS
-      simVars->setFuelAuxLeftPre(leftAuxQuantity);           // in LBS
-      simVars->setFuelAuxRightPre(rightAuxQuantity);         // in LBS
-      simVars->setFuelCenterPre(centerQuantity);             // in LBS
+      simVars->setFuelLeftPre(leftQuantity);          // in LBS
+      simVars->setFuelRightPre(rightQuantity);        // in LBS
+      simVars->setFuelAuxLeftPre(leftAuxQuantity);    // in LBS
+      simVars->setFuelAuxRightPre(rightAuxQuantity);  // in LBS
+      simVars->setFuelCenterPre(centerQuantity);      // in LBS
     } else {
+      if (uiFuelTamper == 1) {
+        fuelLeftPre = leftQuantity;          // LBS
+        fuelRightPre = rightQuantity;        // LBS
+        fuelAuxLeftPre = leftAuxQuantity;    // LBS
+        fuelAuxRightPre = rightAuxQuantity;  // LBS
+        fuelCenterPre = centerQuantity;      // LBS
+      }
       //--------------------------------------------
-      // Left Engine and Wing routine
+      //  Left Engine and Wing routine
```

Comment:
I don't think that's intentional that you added an extra space here?

Reply:
Nope. Thanks! :)